spec/models/journal/aggregated_journal_spec.rb

```diff
@@ -0,0 +1,202 @@
+  ignored_attributes = [:notes_id]
+
+  match do |actual|
+    expected_attributes = get_normalized_attributes expected
+    actual_attributes = get_normalized_attributes actual
+
+    expected_attributes.except(*ignored_attributes) == actual_attributes.except(*ignored_attributes)
+  end
+
+  failure_message do |actual|
+    expected_attributes = get_normalized_attributes expected
+    actual_attributes = actual.attributes.symbolize_keys
+    ["expected attributes: #{display_sorted_hash(expected_attributes.except(*ignored_attributes))}",
+     "actual attributes:   #{display_sorted_hash(actual_attributes.except(*ignored_attributes))}"]
+      .join($/)
+  end
+
+  def get_normalized_attributes(journal)
+    result = journal.attributes.symbolize_keys
+    if result[:created_at]
+      # µs are not stored in all DBMS
+      result[:created_at] = result[:created_at].change(usec: 0)
+    end
+
+    result
+  end
+
+  def display_sorted_hash(hash)
+    '{ ' + hash.sort.map { |k, v| "#{k.inspect}=>#{v.inspect}" }.join(', ') + ' }'
+  end
+end
+
+describe Journal::AggregatedJournal, type: :model do
+  let(:work_package) {
+    FactoryGirl.build(:work_package)
+  }
+  let(:user1) { FactoryGirl.create(:user) }
+  let(:user2) { FactoryGirl.create(:user) }
+  let(:initial_author) { user1 }
+
+  subject { described_class.order('id ASC').to_a }
+
+  before do
+    allow(User).to receive(:current).and_return(initial_author)
+    work_package.save!
+  end
+
+  it 'returns the one and only journal' do
+    expect(subject.count).to eql 1
+    expect(subject.first).to be_equivalent_to_journal work_package.journals.first
+  end
+
+  it 'also indicates its ID via notes_id' do
+    expect(subject.first.notes_id).to eql work_package.journals.first.id
+  end
+
+  it 'is the initial journal' do
+    expect(subject.first.initial?).to be_truthy
+  end
+
+  context 'WP updated immediately after uncommented change' do
+    let(:notes) { nil }
+
+    before do
+      changes = { status: FactoryGirl.build(:status) }
+      changes[:notes] = notes if notes
+
+      expect(work_package.update_by!(new_author, changes)).to be_truthy
+    end
+
+    context 'by author of last change' do
+      let(:new_author) { initial_author }
+
+      it 'returns a single aggregated journal' do
+        expect(subject.count).to eql 1
+        expect(subject.first).to be_equivalent_to_journal work_package.journals.second
+      end
+
+      it 'is the initial journal' do
+        expect(subject.first.initial?).to be_truthy
+      end
+
+      context 'with a comment' do
+        let(:notes) { 'This is why I changed it.' }
+
+        it 'returns a single aggregated journal' do
+          expect(subject.count).to eql 1
+          expect(subject.first).to be_equivalent_to_journal work_package.journals.second
+        end
+
+        context 'adding a second comment' do
+          before do
+            expect(work_package.update_by!(new_author, notes: notes)).to be_truthy
```

Comment:
Using the same comment here threw me off a little. Suggestion: Use a ``notes`` text which smartly explains why it leads to the journals not being merged.

Could this help to understand the situation a little better if the following test were to fail? (Think about what might be written as the reason for test failure)